plugins/module_utils/foreman_helper.py

```diff
@@ -313,14 +313,26 @@ def __init__(self, **kwargs):
         mutually_exclusive = kwargs.pop('mutually_exclusive', []) + [['medium', 'kickstart_repository']]
         super(HostMixin, self).__init__(foreman_spec=foreman_spec, required_plugins=required_plugins, mutually_exclusive=mutually_exclusive, **kwargs)
 
-        if 'activation_keys' in self.foreman_params:
-            if 'parameters' not in self.foreman_params:
-                self.foreman_params['parameters'] = []
-            ak_param = {'name': 'kt_activation_keys', 'parameter_type': 'string', 'value': self.foreman_params.pop('activation_keys')}
-            self.foreman_params['parameters'].append(ak_param)
+    def run(self, **kwargs):
+        entity = self.lookup_entity('entity')
+
+        if not self.desired_absent:
+            if 'activation_keys' in self.foreman_params:
+                if 'parameters' not in self.foreman_params:
+                    parameters = [param for param in (entity or {}).get('parameters', []) if param['name'] != 'kt_activation_keys']
+                else:
+                    parameters = self.foreman_params['parameters']
+                ak_param = {'name': 'kt_activation_keys', 'parameter_type': 'string', 'value': self.foreman_params.pop('activation_keys')}
+                self.foreman_params['parameters'] = parameters + [ak_param]
+            elif 'activation_keys' not in self.foreman_params and 'parameters' in self.foreman_params and entity is not None:
```

Comment:
The first part of that condition is already established by the else in elif.

Reply:
True, that's from an older attempt of mine.

Reply:
```suggestion
            elif 'parameters' in self.foreman_params and entity is not None:
```